src/box/lua/iproto.c

```diff
@@ -268,16 +305,115 @@ push_iproto_protocol_features(struct lua_State *L)
 					  lengthof(features));
 }
 
+/**
+ * Encodes a packet header/body argument to MsgPack: if the argument is a
+ * string, then no encoding is needed — otherwise the argument must be a Lua
+ * table. The Lua table is encoded to MsgPack using IPROTO key translation
+ * table.
+ * In both cases, the result is stored on the fiber region.
+ */
+static const char *
+encode_packet(struct lua_State *L, int idx, size_t *mp_len)
+{
+	int packet_part_type = lua_type(L, idx);
+	struct region *gc = &fiber()->gc;
+	if (packet_part_type == LUA_TSTRING) {
+		const char *arg = lua_tolstring(L, idx, mp_len);
+		char *mp = xregion_alloc(gc, *mp_len);
+		return memcpy(mp, arg, *mp_len);
+	}
+	assert(packet_part_type == LUA_TTABLE);
+	/*
+	 * Can throw a Lua exception, which we cannot catch, and cause the fiber
+	 * region to leak.
+	 */
+	struct mpstream stream;
+	mpstream_init(&stream, gc, region_reserve_cb, region_alloc_cb,
+		      luamp_error, L);
+	size_t used = region_used(gc);
+	/*
+	 * Can throw a Lua exception, which we cannot catch, and cause the fiber
+	 * region to leak. This issue is tracked in #7939.
```

Comment:
Nit: There two very similar comments for the same issue. I'd drop the previous one.

Also, I think we should start comments like this with `TODO(gh-XXXX)` or `FIXME(gh-XXXX)` so that they can be easily grepped.

```
/*
 * FIXME(gh-7939): Can throw a Lua exception ...
 */
```

Reply:
> There two very similar comments for the same issue.

Apparently, these issue are different — one is related to OOM error, while the other is related to encoding error.

Reply:
Fixed. I suggest we track both of the problems in scope of the same issue.